r/tests/testthat/test-dplyr-funcs-string.R

```diff
@@ -248,14 +254,26 @@ test_that("grepl with ignore.case = TRUE and fixed = TRUE", {
   expect_equal(
     df %>%
       Table$create() %>%
-      filter(x = grepl("^B.+", x, ignore.case = TRUE, fixed = TRUE)) %>%
+      filter(grepl("^B.+", x, ignore.case = TRUE, fixed = TRUE)) %>%
       collect(),
     tibble(x = character(0))
   )
+  expect_equal(
+    df %>%
+      Table$create() %>%
+      mutate(
+        a = grepl("O", x, ignore.case = TRUE, fixed = TRUE)
+      ) %>%
+      collect(),
+    tibble(
+      x = c("Foo", "bar", NA_character_),
+      a = c(TRUE, FALSE, FALSE)
+    )
+  )
```

Comment:
```suggestion
  compare_dplyr_binding(
    .input %>%
      mutate(
        a = grepl("O", x, ignore.case = TRUE, fixed = TRUE)
      ) %>%
      collect(),
    df
  )
```

This is more stylistic than anything else, but you should be able to swap out this code

This uses one of our custom expectations, which can be a little hairy at first, but test a few different routes (as a table, as a record batch) + confirm we're getting the same behavior as base R | the tidyverse functions we are binding

https://github.com/apache/arrow/blob/acc6c2eb8ae580085296f608c1a0e8564269456d/r/tests/testthat/helper-expectation.R#L69-L137

Reply:
Aaah, yup you're absolutely right. I missed that comment up there

Reply:
Given that base `grepl` gives that warning when `ignore.case = TRUE` and `fixed = TRUE`, should the binding for `grepl` (as well as for `sub` and `gsub`) also emit that warning?

Reply:
Hmmm, yeah maybe. Would you mind opening a Jira for that (we can get some discussion there + implement it separately so as not to extend the scope of this PR too much!)